.github/scripts/check_labels.py

```diff
@@ -40,21 +39,15 @@ def delete_comment(comment_id: int) -> None:
 def has_required_labels(pr: GitHubPR) -> bool:
     pr_labels = pr.get_labels()
 
-    # Check if PR is not user facing
-    is_not_user_facing_pr = any(label.strip() == "topic: not user facing" for label in pr_labels)
-    if is_not_user_facing_pr:
-        return True
-
-    # Check if bot has already posted a message within the past hour to include a release notes label
+    # Delete all previous comments
     for comment in pr.get_comments():
         if comment.body_text.lstrip(" #").startswith(ERR_MSG_TITLE) and comment.author_login in BOT_AUTHORS:
-            ts = datetime.strptime(comment.created_at, "%Y-%m-%dT%H:%M:%SZ")
-            if (datetime.utcnow() - ts) < timedelta(hours=1):
```

Comment:
This old check was there to rate limit ourselves if a person is pushing multiple commits to their PR but they haven't yet added a label it it.

Perhaps it's better to split this into two functions:
- has_required_labels(pr) - which only returns a bool and doesn't delete any comments
- delete_stale_comments(pr, has_required_labels: bool) - If the labels are there, then it deletes it's old comment. If the label still isn't there, then it'll only delete the comment if it was last posted more than an hour ago (the old behavior)

Reply:
* split up into check for labels, add comment, and delete comment
* delete comments if has labels
* add comment if does not have label and does not have comment yet - not quite the same but i dont want to spam with extra comments when one already exists